webapp/Connector/src/window/MabGridFacet.js

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright (c) 2016-2017 LabKey Corporation
+ *
+ * Licensed under the Apache License, Version 2.0: http://www.apache.org/licenses/LICENSE-2.0
+ */
+Ext.define('Connector.window.MabGridFacet', {
+
+    extend: 'Connector.window.AbstractGroupedFacet',
+
+    filterConfig: undefined,
+
+    currentfieldName: '',
+
+    blankValue: '[blank]',
+
+    /* To avoid URL overflow, allow up to 100 selections per column */
+    maxSelection: 100,
+
+    initComponent : function()
+    {
+        this.callParent();
+        this.addEvents('mabfiltersearchchanged');
+        this.createFacetGrid();
+        this.on('mabfiltersearchchanged', this.facetGrid.filterFacetOptions, this.facetGrid);
+    },
+
+    // Override to add search box
+    getTopConfig : function() {
```

Comment:
If I'm reading this correctly, this search functionality only exists for the Mab facet filter window. Search seems like a valuable thing to add more generally to facet filter windows, but that is likely a future thing.